pandas/io/rdata.py

```diff
@@ -0,0 +1,598 @@
+import io
+import os
+from tempfile import TemporaryDirectory
+from typing import (
+    Dict,
+    List,
+    Optional,
+    Union,
+)
+
+from pandas._typing import (
+    Buffer,
+    CompressionOptions,
+    FilePathOrBuffer,
+    StorageOptions,
+)
+from pandas.compat._optional import import_optional_dependency
+from pandas.errors import AbstractMethodError
+from pandas.util._decorators import doc
+
+from pandas.core.dtypes.common import is_list_like
+
+from pandas.core.frame import DataFrame
+from pandas.core.shared_docs import _shared_docs
+
+from pandas.io.common import (
+    file_exists,
+    get_handle,
+    is_fsspec_url,
+    is_url,
+    stringify_path,
+)
+
+
+@doc(storage_options=_shared_docs["storage_options"])
+def read_rdata(
+    path_or_buffer: FilePathOrBuffer,
+    file_format: str = "infer",
+    select_frames: Optional[List[str]] = None,
+    rownames: bool = True,
+    storage_options: StorageOptions = None,
+) -> Union[DataFrame, Dict[str, DataFrame]]:
```

Comment:
hmm, shouldn't this *always* return a dict-of-frames? e.g. when does it and when does it not?